homeassistant/components/calendar/__init__.py

```diff
@@ -0,0 +1,286 @@
+
+
+# pylint: disable=too-many-instance-attributes
+class CalendarEventDevice(Entity):
+    """A calendar event device."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, calendar, dev_id, data):
+        """Create the Calendar Event Device.
+
+        hass = HA instance
+        calendar = reference for subclasses to get main cal object
+        data = dict of track/search/name/offset
+        """
+        self.hass = hass
+        self._state = False
+        self._calendar = calendar
+        self._track = data.get('track', DEFAULT_CONF_TRACK_NEW)
+        self._search = data.get('search', None)
+        self._name = data.get('name')
+        self.dev_id = dev_id
+        self._offset = data.get('offset', DEFAULT_CONF_OFFSET)
+        self.entity_id = generate_entity_id(ENTITY_ID_FORMAT,
+                                            self.dev_id,
+                                            hass=hass)
+        self._start_listener = None
+        self._end_listener = None
+
+        self._cal_data = {
+            'all_day': False,
+            'offset_time': dt.dt.timedelta(),
+            'message': '',
+            'start': None,
+            'end': None,
+            'location': None,
+            'description': '',
+        }
+
+        if self._track:
+            self.update()
+
+    @property
+    def offset_reached(self):
+        """Have we reached the offset time specified in the event title."""
+        if self._cal_data['start'] != None and \
+           self._cal_data['offset_time'] == dt.dt.timedelta():
+            return False
+
+        return self._cal_data['start'] + self._cal_data['offset_time'] <= \
+            dt.now(self._cal_data['start'].tzinfo)
+
+    @property
+    def state_attributes(self):
+        """State Attributes for HA."""
+        start = self._cal_data.get('start', None)
+        end = self._cal_data.get('end', None)
+        start = start.isoformat() if start is not None else None
+        end = end.isoformat() if end is not None else None
+
+        return {
+            'message': self._cal_data.get('message', ''),
+            'all_day': self._cal_data.get('all_day', False),
+            'offset_reached': self.offset_reached,
+            'start_time': start,
+            'end_time': end,
+            'location': self._cal_data.get('location', None),
+            'description': self._cal_data.get('description', None),
+        }
+
+    @property
+    def track(self):
+        """Are we tracking events on this search sensor."""
+        return self._track
+
+    @property
+    def is_all_day(self):
+        """If the next event is all day then True."""
+        return self._cal_data.get('all_day', False)
+
+    @property
+    def search(self):
+        """Return the search string."""
+        return self._search
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._name
+
+    @property
```

Comment:
I'm confused. I thought this was needed so I could check it in an automation?
```
- condition: state
  entity_id: calendar.name
  state: 'on'
```

I left it as on/off since it made sense to me as these are basically binary switches for being in an event state.